languages/julia/exercises/concept/poetry-club/.meta/test_generator_util.jl

```diff
@@ -0,0 +1,249 @@
+# Writing out the tests explicitely avoids giving away the solution in the testsuite.
+# It also leads to a minimal number of prerequisites.
+# This file can be used to generate them.
+
+const tab = "    "
+
+include("example.jl")
+
+struct Poem
+    title::String
+    body::String
+    author::String
+    acrostic::Bool
+    telestich::Bool
+end
+
+const poems = (
+    Poem("Shire",
+    
+        """
+        Stands so high
+        Huge hooves too
+        Impatiently waits for
+        Reins and harness
+        Eager to leave""",
+
+        "Michael Lockwood",
+        true,
+        true
+    ),
+
+    Poem("Summer",
+        
+        """
+        Sunshine warming my toes,
+        Underwater fun with my friends.
+        Making homemade ice cream on the porch,
+        Many long nights catching fireflies.
+        Early morning walks to the creek,
+        Reveling in the freedom of lazy days.""",
+
+        "John Albert Caballero",
+        true,
+        false
+    ),
+
+    Poem("Sophia",
+
+        """
+        Serene, calming quality
+        Organized, you always have it together
+        Picturesque, strikingly beautiful
+        Honest, so genuine
+        Imaginative, a creative mind
+        Alluring, so attractive""",
+
+        "John Albert Caballero",
+        true,
+        false
+    ),
+
+    Poem("Code",
+
+        """
+        Compilers intensily bestow
+        On commencing without ego
+        Different processes ajar
+        Exit with zero quick""",
+
+        "Derk-Jan Karrenbeld",
+        true,
+        true
+    ),
+
+    Poem("",
+    
+        """
+        Ιησούς
+        Χριστός
+        Θεού
+        Υἱός
+        Σωτήρ""",
+
+        "",
+        true,
+        false
+    ),
+
+    Poem("Колокол",
+    
+        """
+        Произнося чудесный чистый звук,
+        Вишу на колокольне. Высоко!
+        Неоднократно сам звенеть хотел,
+        Разлиться песней сердца далеко,
+        Но мой язык во власти чьих-то рук.
+        Вздохнул бы я свободно и легко,
+        Когда бы сам, не по заказу, пел.""",
+
+        "Иван Чудасов",
+        false,
+        true
+    )
```

Comment:
Source: https://ru.wikipedia.org/wiki/%D0%A2%D0%B5%D0%BB%D0%B5%D1%81%D1%82%D0%B8%D1%85#%D0%98%D1%81%D1%82%D0%BE%D1%80%D0%B8%D1%8F_%D0%B8_%D0%BF%D1%80%D0%B8%D0%BC%D0%B5%D1%80%D1%8B